src/window/basic_frame.rs

```diff
@@ -395,17 +389,13 @@ impl Frame for BasicFrame {
             return;
         }
         let (width, height) = *(self.inner.size.lock().unwrap());
-        let mut buffers = self.buffers.lock().unwrap();
-        let mut current_pool = self.current_pool.lock().unwrap();
-        // destroy currently pending buffers
-        for b in buffers.get_mut(*current_pool).unwrap().drain(..) {
-            b.destroy();
-        }
 
         {
             // grab the current pool
-            let mut pools = self.pools.lock().unwrap();
-            let pool = pools.pool();
+            let pool = match self.pools.pool() {
+                Some(pool) => pool,
+                None => return,
+            };
```

Comment:
Here, to avoid completely missing a draw, there should be an internal boolean flag of the `Frame` that tracks that a draw was skipped because both mempools are in use. This flag would be reset to `false` whenever a redraw is really done.

Using the previously described callback of `DoubleMemPool`, whenever this callback fires, we check the flag in `Frame`. If it is set to `true`, that means that a draw was missed because the memory pools were in use. In this case, the `Frame` triggers a `FrameRequest::Refresh`, to ask the client to trigger a new redraw of the frame, to compensate for the one previously missed.

Reply:
I implemented the frame's DoubleMemPool so that it calls a Refresh event without an internal boolean. This can be done because DoubleMemPool only fires when previously a None was returned from .pool() (which means you have to skip a draw) and a pool is now free. Am I right in thinking that?

Reply:
Setting up `DoubleMemPool` to only fire its event if a `None` was previously returned from `pool()` sounds good, as long as it's clearly documented :+1: